services/dhcp/src/main/java/org/openthinclient/service/dhcp/RemotedBean.java

```diff
@@ -27,20 +27,14 @@
 public class RemotedBean implements Remoted {
 	private static final Logger logger = LoggerFactory.getLogger(RemotedBean.class);
 
-	public boolean dhcpReloadRealms() throws Exception {
-//		final ObjectName objectName = new ObjectName("tcat:service=ConfigService");
-//		final MBeanServer server = MBeanServerFactory.findMBeanServer(null).get(0);
-//
-//		if (Boolean.FALSE.equals(server.invoke(objectName, "reloadRealms",
-//				new Object[]{}, new String[]{}))) {
-//			logger.error("Unable to reloadRealms");
-//			return false;
-//		} else
-//			return true;
-
-    // FIXME this implementation has to be adapted to the new service control mechanism
-
-    throw new UnsupportedOperationException();
-
-  }
+	@Override
+	public boolean dhcpReloadRealms(DhcpService dhcpService) throws Exception {
```

Comment:
Don't provide the DHCP Service as a method parameter. Provide it as a _constructor_ parameter and assign it to a field in the class.